File: Python/Program/Class/Processor.py
from Class.PartKomputer import PartKomputer

class Processor(PartKomputer):
    def __init__(self, id=0, nama="", harga=0, stok=0,
                 no_part="", manufaktur="", kondisi="",
                 socket="", core=0, thread=0, tdp=0):
        super().__init__(id, nama, harga, stok, no_part, manufaktur, kondisi)
        self._socket = socket
        self._core = core
        self._thread = thread
        self._tdp = tdp

    def set_socket(self, socket):
        self._socket = socket

    def get_socket(self):
        return self._socket

    def set_core(self, core):
        self._core = core

    def get_core(self):
        return self._core

    def set_thread(self, thread):
        self._thread = thread

    def get_thread(self):
        return self._thread

    def set_tdp(self, tdp):
        self._tdp = tdp

    def get_tdp(self):
        return self._tdp

File: Java/Program/Class/Processor.java
package Class;

public class Processor extends PartKomputer {
    private String socket;
    private int core;
    private int thread;
    private int TDP;

    public Processor() {
        super();
        this.socket = "";
        this.core = 0;
        this.thread = 0;
        this.TDP = 0;
    }

    public Processor(int id, String nama, int harga, int stok,
                     String noPart, String manufaktur, String kondisi,
                     String socket, int core, int thread, int TDP) {
        super(id, nama, harga, stok, noPart, manufaktur, kondisi);
        this.socket = socket;
        this.core = core;
        this.thread = thread;
        this.TDP = TDP;
    }

    public String getSocket() { return socket; }
    public void setSocket(String socket) { this.socket = socket; }

    public int getCore() { return core; }
    public void setCore(int core) { this.core = core; }

    public int getThread() { return thread; }
    public void setThread(int thread) { this.thread = thread; }

    public int getTDP() { return TDP; }
    public void setTDP(int TDP) { this.TDP = TDP; }
}
